test/extensions/transport_sockets/tls/ssl_socket_test.cc

```diff
@@ -3870,6 +3862,7 @@ TEST_P(SslSocketTest, ProtocolVersions) {
   // so enable them to avoid false positives.
   client_params->add_cipher_suites("ECDHE-RSA-AES128-SHA");
   server_params->add_cipher_suites("ECDHE-RSA-AES128-SHA");
+  updateFilterChain(tls_context, filter_chain);
```

Comment:
Out of curiosity, could this be effectively done inside `testUtilV2`?

Reply:
Thanks for bring this up.
 I would say practically yes,  but technically if we do that, there will be a  implicit assumption:  The created `TestUtilOptionsV2` object  will be only used in `testUtilV2 `function(which I think it is the case now). Because technically this update here is actually used to update `TestUtilOptionsV2`'s listener filter chain. If in the future, we have different test function over `TestUtilOptionsV2` object we are going to miss this update.

My current strategy is whenever I see the updates on server args, I call this update function. (the update will be reflected in `TestUtilOptionsV2` since `listener` is stored as reference member).

Just my 2 cents:  technically a better approach might be putting all server side updates in one place with this `updateFilterChain`. But I feel that will be large amount of refactor (just lots of places:) that could be labeled as TODO if we want to pursue this path.
Please let me know what do you think.

Reply:
That's fine. I don't feel that strongly on this one, it's already somewhat messy.